pkg/controllers/job/job_controller_actions.go

```diff
@@ -129,6 +124,11 @@ func (cc *jobcontroller) killJob(jobInfo *apis.JobInfo, podRetainPhase state.Pha
 		}
 	}
 
+	// Update running duration
+	runningDuration := metav1.Duration{Duration: job.Status.State.LastTransitionTime.Sub(jobInfo.Job.CreationTimestamp.Time)}
```

Comment:
Can `job.Status.State.LastTransitionTime` indicate the job complete time?

Reply:
We modify `LastTransitionTime` inside the if statement above when we receive an update status. Given that we only call `killJob` when transitioning to a terminal state or when restarting the controller, `LastTransitionTime` will be either:

1. The current time if there was a state change, or
2. The time when we last transitioned states, which will be whenever `killJob` was originally called for that job.

Unless I'm missing some other scenario, `LastTransitionTime` doesn't change after a job has been killed, thus it should hold the time when it transitioned to a completed state.

Reply:
seems no problem: )